pkg/controllers/vdb/upgrade.go

```diff
@@ -192,6 +192,15 @@ func (i *UpgradeManager) finishUpgrade(ctx context.Context, sbName string) (ctrl
 	return ctrl.Result{}, nil
 }
 
+// saveRestorePoint handles condition status and event recording for start of an upgrade
+func (i *UpgradeManager) saveRestorePoint(ctx context.Context) (ctrl.Result, error) {
```

Comment:
@roypaulin Follow up question as https://github.com/vertica/vertica-kubernetes/pull/920/files#r1765057129, should we maintain the saveRestore logic for upgrade in one place. So in the offline/readonly/online reconciler we can call o.Manager.saveRestorePoint()